frontend/src/routes/(authenticated)/(dashboards)/admin/+page.svelte

```diff
@@ -50,44 +62,93 @@
   let projectSearch = '';
   let userSearch = '';
   $: projectSearchLower = projectSearch.toLocaleLowerCase();
+  $: projectLimit = projectSearch ? Infinity : 10;
   $: projects = $allProjects
     .filter(
       (p) =>
         !projectSearch ||
         p.name.toLocaleLowerCase().includes(projectSearchLower) ||
         p.code.toLocaleLowerCase().includes(projectSearchLower)
     )
-    .slice(0, projectSearch ? undefined : 10);
+    .slice(0, projectLimit);
   $: userSearchLower = userSearch.toLocaleLowerCase();
+  $: userLimit = userSearch ? Infinity : 10;
   $: users = $allUsers
     .filter(
       (u) =>
         !userSearch ||
         u.name.toLocaleLowerCase().includes(userSearchLower) ||
         u.email.toLocaleLowerCase().includes(userSearchLower)
     )
-    .slice(0, userSearch ? undefined : 10);
+    .slice(0, userLimit);
+
+  async function softDeleteProject(project: (typeof projects)[0]): Promise<void> {
+    const result = await deleteProjectModal.open(project.name, async () => {
+      const { error } = await _deleteProject(project.id);
+      return error?.message;
+    });
+    if (result.response === DialogResponse.Submit) {
+      notifyWarning($t('delete_project_modal.success', { name: project.name, code: project.code }));
+    }
+  }
+
+  async function refreshPage(): Promise<void> {
+    await goto(`?${toSearchParams(options)}`, {
```

Comment:
I wonder if we were to turn options into a custom store, then whenever a value changes we do a goto like this in a generic way. Maybe not in this PR since it's big already though.

Reply:
That would be nifty. It would probably mean that we can do simple binding like: `bind:checked={options.showDeletedProjects}`. Anyway, yeah, I won't do anything in that direction in this PR.